raiden/channel/balance_proof.py

```diff
@@ -209,30 +209,38 @@ def get_known_unlocks(self):
             self.hashlock_unlockedlocks.itervalues(),
         )
 
+        tree = self.generate_merkle_tree();
+
         return [
             self.compute_proof_for_lock(
                 partialproof.secret,
                 partialproof.lock,
+                tree,
             )
             for partialproof in allpartialproof
         ]
 
-    def compute_proof_for_lock(self, secret, lock):
-        alllocks = chain(
-            self.hashlock_pendinglocks.values(),
-            self.hashlock_unclaimedlocks.values(),
-            self.hashlock_unlockedlocks.values()
-        )
+    def compute_proof_for_lock(self, secret, lock, tree=None):
```

Comment:
Hmm you have a point there.

I don't like the optional argument approach myself which is why I suggested the alternative but for consistency's sake perhaps keep it as is and we can address in some other PR.

Just remove the semicolon and we can merge :)